libcaf_io/src/interfaces.cpp

```diff
@@ -276,6 +278,32 @@ interfaces::server_address(uint16_t port, const char* host,
   return results;
 }
 
+bool interfaces::get_endpoint(const std::string& host, uint16_t port,
+                              ip_endpoint& ep,
+                              optional<protocol::network> preferred) {
+  addrinfo hint;
+  char port_hint[6];
+  sprintf(port_hint, "%hu", port);
```

Comment:
I would use `snprintf`, but since port is `uint16_t`, this should work...

Reply:
The main problem is that `%hu` identifies an `unsigned short int`. So this code silently assumes that `sizeof(uint16_t) == sizeof(unsigned short int)`. The magic number 6 is also not motivated. I assume this is because 2^16 = 65535, which is 5 characters at the most + a null terminator.

I think at least a safeguard in form of a `static_assert` should be added here.